loaders/embededKeyLoader_test.go

```diff
@@ -0,0 +1,314 @@
+		require.NoError(t, err)
+		assert.NotNil(t, key)
+	})
+
+	t.Run("AtomicQueryV3CircuitID - beta.1", func(t *testing.T) {
+		loader := NewEmbeddedKeyLoader()
+		key, err := loader.Load(circuits.AtomicQueryV3CircuitID)
+		require.NoError(t, err)
+		assert.NotNil(t, key)
+	})
+
+	t.Run("AtomicQueryV3OnChainCircuitID - beta.1", func(t *testing.T) {
+		loader := NewEmbeddedKeyLoader()
+		key, err := loader.Load(circuits.AtomicQueryV3OnChainCircuitID)
+		require.NoError(t, err)
+		assert.NotNil(t, key)
+	})
+
+	t.Run("LinkedMultiQuery10CircuitID - beta.1", func(t *testing.T) {
+		loader := NewEmbeddedKeyLoader()
+		key, err := loader.Load(circuits.LinkedMultiQuery10CircuitID)
+		require.NoError(t, err)
+		assert.NotNil(t, key)
+	})
+}
+
+func TestEmbeddedKeyLoader_Load_EmbeddedKeys(t *testing.T) {
+	t.Run("load embedded key", func(t *testing.T) {
+		loader := NewEmbeddedKeyLoader()
+		key, err := loader.Load(circuits.AuthV2CircuitID)
+		require.NoError(t, err)
+		assert.NotNil(t, key)
+	})
+
+	t.Run("embedded key not found", func(t *testing.T) {
+		loader := NewEmbeddedKeyLoader()
+		_, err := loader.Load("non-existent-circuit")
+		assert.Error(t, err)
+		assert.Contains(t, err.Error(), "failed to load default key")
+	})
+}
+
+func TestEmbeddedKeyLoader_Load_CustomLoaderError(t *testing.T) {
+	t.Run("custom loader error", func(t *testing.T) {
+		mockLoader := &MockKeyLoader{
+			err: errors.New("mock error"),
+		}
+		loader := NewEmbeddedKeyLoader(WithKeyLoader(mockLoader))
+
+		_, err := loader.Load(circuits.CircuitID("any-circuit"))
+		assert.Error(t, err)
+		assert.Equal(t, "mock error", err.Error())
+	})
+}
+
+func TestEmbeddedKeyLoader_Cache(t *testing.T) {
+	t.Run("cache enabled", func(t *testing.T) {
+		loader := NewEmbeddedKeyLoader()
+		testID := circuits.CircuitID("test-circuit")
+		testKey := []byte("test-key-data")
+
+		loader.storeInCache(testID, testKey)
+		cachedKey := loader.getFromCache(testID)
+		assert.Equal(t, testKey, cachedKey)
+	})
+
+	t.Run("cache disabled", func(t *testing.T) {
+		loader := NewEmbeddedKeyLoader(WithCacheDisabled())
+		testID := circuits.CircuitID("test-circuit")
+		testKey := []byte("test-key-data")
+
+		loader.storeInCache(testID, testKey)
+		cachedKey := loader.getFromCache(testID)
+		assert.Nil(t, cachedKey)
+	})
+}
+
+func TestFSKeyLoader_Load_KeyNotFound(t *testing.T) {
+	loader := FSKeyLoader{Dir: "/non/existent/path"}
+
+	_, err := loader.Load(circuits.CircuitID("non-existent-circuit"))
+	assert.Error(t, err)
+	assert.Equal(t, ErrKeyNotFound, err)
+}
+
+func TestEmbeddedKeyLoader_CacheConcurrency(t *testing.T) {
+	loader := NewEmbeddedKeyLoader()
+	testID := circuits.CircuitID("test-circuit")
+	testKey := []byte("test-key-data")
+
+	// Test concurrent reads
+	t.Run("concurrent reads", func(t *testing.T) {
+		loader.storeInCache(testID, testKey)
+
+		var wg sync.WaitGroup
+		for i := 0; i < 100; i++ {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				key := loader.getFromCache(testID)
```

Comment:
In my opinion, this test is not useful. Since you are testing parallel reads without write operations, there is no concurrency issue.